sycl/doc/extensions/DeviceGlobal/SPV_INTEL_global_variable_decorations.asciidoc

```diff
@@ -0,0 +1,224 @@
+= SPV_INTEL_global_variable_decorations
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+// This is necessary for asciidoc, but not for asciidoctor
+:cpp: C++
+
+== Name Strings
+
+SPV_INTEL_global_variable_decorations
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Headers
+
+== Contributors
+
+Artem Radzikhovskyy, Intel +
+Michael Kinsner, Intel +
+Joe Garvey, Intel +
+Mohammad Fawaz, Intel +
+Gregory Lueck, Intel
+
+== Notice
+
+Copyright (c) 2021 Intel Corporation.  All rights reserved.
+
+== Status
+
+Final Draft
+
+== Version
+
+Built On: {docdate} +
+Revision: 2
+
+== Dependencies
+
+This extension is written against the SPIR-V Specification,
+Version 1.5 Revision 5.
+
+This extension requires SPIR-V 1.0.
+
+== Overview
+
+This extension adds decorations that can be applied to global (module scope)
+variables.  These decorations are mostly intended to help code generation for
+FPGA devices, but they must be recognized by any consumer that supports this
+extension.
+
+== Extension Name
+To use this extension within a SPIR-V module, the following *OpExtension* must
+be present in the module:
+
+----
+OpExtension "SPV_INTEL_global_variable_decorations"
+----
+
+== New Capabilities
+This extension introduces a new capability:
+
+----
+GlobalVariableDecorationsINTEL
+----
+
+== Token Number Assignments
+
+--
+[width="40%"]
+[cols="70%,30%"]
+[grid="rows"]
+|====
+|GlobalVariableDecorationsINTEL | 6146
+|HostAccessINTEL | 6147
+|InitModeINTEL | 6148
+|ImplementInCSRINTEL | 6149
+|====
+--
+
+== Modifications to the SPIR-V Specification, Version 1.5
+
+=== Decoration
+
+Modify Section 3.20, Decoration, adding these rows to the Decoration table:
+
+--
+[options="header"]
+|====
+2+^| Decoration 2+^| Extra Operands | Enabling Capabilities
+
+// --- ROW BREAK ---
+| 6147
+a|
+*HostAccessINTEL* +
+Only valid on global (module scope) *OpVariable*.
+
+The client API's execution environment may provide a way to access a global
+variable's value from the host system.  If it does, this decoration provides
+two pieces of information.  _Access_ is an assertion by the producer about the
+types of these accesses, which may allow the consumer to perform certain
+optimizations, especially for FPGA devices.  _Name_ is a name which the client
+API's execution environment may use to identify this variable.
+
+Legal values of _Access_:
+
+* 0 [Read] - The execution environment may read the variable from the host but
+  will never write it.  On an FPGA device, only a read memory port is exposed
+  by the generated IP core.
```

Comment:
Should we say "kernel" instead of IP core?  IP core isn't a term that is defined in the SPIR-V spec.  If we make this change it should apply to the other 3 access modes as well.

Reply:
Or we could say "SPIR-V module" like we do below for the ImplementInCSRINTEL control.

Reply:
Neither "kernel" nor "SPIR-V module" seem right to me.  I think this sentence is talking about how this decoration affects the RTL that is generated for an FPGA device, so it makes sense that we are not using a term that is defined in the SPIR-V spec.

I think I did not originally suggest this wording about "IP core".  Maybe it came from @artemrad?

One option is to just say this, which matches the wording we use in the API specification for "device global":

> For an FPGA device, only a read port is exposed.

Does that sound better?

Reply:
Yes, I think I like that more.